Sources/Sentry/SentryProfiler.mm

```diff
@@ -574,19 +574,38 @@ - (void)captureEnvelope
 
     // populate info from all transactions that occurred while profiler was running
     auto transactionsInfo = [NSMutableArray array];
+    SENTRY_LOG_DEBUG(@"Profile start timestamp: %@ absolute time: %llu", _startDate,
+        (unsigned long long)_startTimestamp);
+    SENTRY_LOG_DEBUG(@"Profile end timestamp: %@ absolute time: %llu", _endDate,
+        (unsigned long long)_endTimestamp);
     for (SentryTransaction *transaction in _transactions) {
+        SENTRY_LOG_DEBUG(@"Transaction %@ start timestamp: %@", transaction.trace.context.traceId,
+            transaction.startTimestamp);
+        SENTRY_LOG_DEBUG(@"Transaction %@ end timestamp: %@", transaction.trace.context.traceId,
+            transaction.timestamp);
         const auto relativeStart =
             [NSString stringWithFormat:@"%llu",
                       [transaction.startTimestamp compare:_startDate] == NSOrderedAscending
                           ? 0
                           : (unsigned long long)(
                               [transaction.startTimestamp timeIntervalSinceDate:_startDate] * 1e9)];
-        const auto relativeEnd =
-            [NSString stringWithFormat:@"%llu",
-                      [transaction.timestamp compare:_endDate] == NSOrderedDescending
-                          ? profileDuration
-                          : (unsigned long long)(
-                              [transaction.timestamp timeIntervalSinceDate:_startDate] * 1e9)];
+
+        NSString *relativeEnd;
```

Comment:
`m`: Please add some tests for this new functionality.

Reply:
Improved the test here to ensure it doesn't get the same value that will cause it to fail in backend validation: https://github.com/getsentry/sentry-cocoa/pull/2359/commits/11bc3962303af67a06999e419173a33dd1e5f7e1